cucumber-rails.gemspec

```diff
@@ -22,11 +22,11 @@ Gem::Specification.new do |s|
     'source_code_uri' => "https://github.com/cucumber/cucumber-rails/tree/v#{s.version}"
   }
 
-  s.add_runtime_dependency('capybara', ['>= 2.12', '< 4'])
-  s.add_runtime_dependency('cucumber', ['>= 3.0.2', '< 5'])
-  s.add_runtime_dependency('mime-types', ['>= 2.0', '< 4'])
-  s.add_runtime_dependency('nokogiri', '~> 1.8')
-  s.add_runtime_dependency('rails', ['>= 5.0', '< 7'])
+  s.add_runtime_dependency('capybara', ['>= 2.18', '< 4']) # We support legacy capybara (But only the last 2.x)
+  s.add_runtime_dependency('cucumber', ['>= 3.0.2', '< 6']) # Support cucumber in the 3.x / 4.x / 5.x revisions
+  s.add_runtime_dependency('mime-types', ['~> 3.2']) # Only support the latest major (3+ years old)
+  s.add_runtime_dependency('nokogiri', '~> 1.8') # Only support the latest major (3+ years old)
+  s.add_runtime_dependency('rails', ['>= 5.0', '< 7']) # We support any version of Rails in the 5.x and 6.x series
```

Comment:
Maybe it's clearer if we put one comment at the top here stating we support the last major version older than three years, or the version from the oldest support release of Rails. In any case, stating the general rule at the top would be nice.

If we want to support capybara 2 we should have at least one Appraisal that includes it.

Reply:
These comments were added from a suggestion by Aslak that we document gems that don't use our auto-update everything script.

Each gem support is different purely because we've not got a huge consumer base and I guess I'm trying to update them slowly.

Could state the general rule at top, happy to do that.

Agree we should probably have a capybara 2 use-case if we're still supporting it (Which we need to do for now).